### DISCUSSION TITLE
Will IRIS shaders support internal shaders from texture packs?

### DISCUSSION THREAD

**Discussion by Commandthealien** [IrisShaders/Iris#1841] [Ideas]
I think if Texture Packs want to use their own internal shaders it should be allowed/possible right.

Like what @TheNukeLore said in this post https://github.com/IrisShaders/Iris/discussions/1666#discussioncomment-3783020_

"This is something that we want to avoid as internal shaders should never be used, there are external mods that give the same affect but way more performant and cause less bugs."

So I assume there won't be compatibility with texture pack internal shaders?

**Comment by TheNukeLore** [comment]
It’s not possible for texturepack(core) shaders to be used with iris/optifine shaders, it’s just not reasonably possible.

**Comment by IMS212** [comment]
No, as there are technical limits. See our documentation for details.
